skopt/learning/gaussian_process/gpr.py

```diff
@@ -224,10 +229,10 @@ def fit(self, X, y):
         self.K_inv_ = L_inv.dot(L_inv.T)
 
         # Fix deprecation warning #462
-        if int(sklearn.__version__[2:4]) >= 23:
+        if _SKLEARN_VERSION >= _V_0_23:
             self.y_train_std_ = self._y_train_std
             self.y_train_mean_ = self._y_train_mean
-        elif int(sklearn.__version__[2:4]) >= 19:
+        elif _SKLEARN_VERSION >= _V_0_19:
```

Comment:
If you rather simply do:
```py
if sklearn.__version__ >= '0.23':
    ...
if sklearn.__version__ >= '0.19':
    ...
```
we won't require a dependency on package [_packaging_](https://pypi.org/project/packaging/).

Reply:
I'm reluctant to use string comparison instead of proper version comparison, since bad version comparison is the source of #1061 .

I incorrectly though `scikit-optimize` indirectly depended on `packaging` already, but this is not the case (`packaging` is required by `pytest`).

Apparently both [`scipy`](https://github.com/scipy/scipy/blob/master/scipy/_lib/_pep440.py) and [`scikit-learn`](https://github.com/scikit-learn/scikit-learn/pull/19826) chose to vendor `packaging.parse_version` for this purpose. Maybe this is the best route ? Or maybe we can use their vendored version, since `scikit-optimize` depends on both ? I think this will increase the minimum supported `scikit-learn` version, I have a vague memory that older versions didn't vendor it.

Reply:
I _don't_ think this is the best route. Extra dependencies for a two-line fix rarely are.

Do you see a problem with the above proposal? It's a common idiom. Previously it was implemented incorrectly.

Reply:
I think scipy and scikit-learn copy the code of the single module they need to avoid the dependency.
I don't really see a problem with the above idiom, I'm just not sure if alphabetical ordering matches version ordering. Since it is a common idiom, it should be robust enough, I'll go with that.

Reply:
sklearn 0.2 and 0.20 are far enough apart for us to not care. :smile: 
It should be good enough for this release. Which is kinda urgent.

Reply:
That's fair, let's not overcomplicate that